frontend/src/components/CalcSpectrum.tsx

```diff
@@ -152,22 +152,53 @@ export const CalcSpectrum: React.FC = () => {
       species: data.species,
     });
     import(/* webpackIgnore: true */ "./config.js").then(async (module) => {
-      const rawResponse = await axios.post(module.apiEndpoint + endpoint, data);
-      if (
-        rawResponse.data.data === undefined &&
-        rawResponse.data.error === undefined
-      ) {
-        handleBadResponse("Bad response from backend!");
-      } else {
-        const response = await rawResponse.data;
-        if (response.error) {
-          handleBadResponse(response.error);
+      if (endpoint === "calculate-spectrum") {
+        console.log("calculate-spectrum");
+        const rawResponse = await axios.post(
+          module.apiEndpoint + endpoint,
+          data
+        );
+        if (
+          rawResponse.data.data === undefined &&
+          rawResponse.data.error === undefined
+        ) {
+          handleBadResponse("Bad response from backend!");
         } else {
-          setCalcSpectrumResponse(response);
+          const response = await rawResponse.data;
+          if (response.error) {
+            handleBadResponse(response.error);
+          } else {
+            setCalcSpectrumResponse(response);
+          }
         }
+        setLoading(false);
+      }
+      if (endpoint === "download-spectrum") {
+        setLoading(false);
+        console.log("Download spectrum");
+        await axios({
+          url: module.apiEndpoint + `download-spectrum`,
+          method: "POST",
+          responseType: "blob",
+          data: data,
+          headers: {
+            "Content-Type": "application/json",
+          },
+        })
+          .then((response) => {
+            const url = window.URL.createObjectURL(new Blob([response.data]));
+            const link = document.createElement("a");
+            link.href = url;
+            link.setAttribute("download", "radis.spec");
+            document.body.appendChild(link);
+            link.click();
+          })
+          .catch((error) => {
+            handleBadResponse(error);
+          });
```

Comment:
It's considered to be a better style to fully use either async/await or then(). Right now we are using async/await else where in the code (for example await axios(...)), so it would be nice to remove then then() and catch() to additionally be async/await since that is the more modern style.

Reply:
Please change from then() to async/await syntax to match other code in the repo.

Additionally, I we need "handleBadResponse" to check for errors like when hitting calc-spectrum as right now backend errors will not be shown to the user. We need to catch errors in the response like this: https://github.com/suzil/radis-app/pull/619/files#diff-c16fbf0c6f7b90a46b94b36f88893c2d174476088608841f7254afba0e81373dR156